nx/test/nx/defn_test.exs

```diff
@@ -1126,19 +1160,19 @@ defmodule Nx.DefnTest do
                cond_branch_elimination(Nx.tensor([1, 2, 3]), Nx.tensor(0))
     end
 
-    test "raises if cond is missing last atom clause" do
-      assert_raise CompileError, ~r"expected the last clause of cond to match on an atom", fn ->
-        defmodule InvalidCond do
-          defn badcond(a) do
-            cond do
-              Nx.any(a) -> +a
-              Nx.all(a) -> -a
-            end
-          end
-        end
+    defn bad_cond(a) do
+      cond do
+        Nx.any(a) -> +a
+        Nx.all(a) -> -a
       end
     end
 
+    test "raises if cond is missing last atom clause" do
```

Comment:
Should `raise` inside the condition already be supported here? If so, we should test on that as well

Reply:
Done!